src/test/java/kitchenpos/bo/TableBoTest.java

```diff
@@ -0,0 +1,117 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.BDDMockito.given;
+
+/**
+ * @author Geonguk Han
+ * @since 2020-02-15
+ */
+@ExtendWith(MockitoExtension.class)
+class TableBoTest extends Fixtures {
+
+    @InjectMocks
+    private TableBo tableBo;
+
+    @Mock
+    private OrderDao orderDao;
+
+    @Mock
+    private OrderTableDao orderTableDao;
+
+    @Test
+    @DisplayName("주문 테이블을 등록할 수 있다.")
+    void create() {
+        given(orderTableDao.save(orderTable)).willReturn(orderTable);
+
+        final OrderTable savedOrderTable = tableBo.create(orderTable);
+
+        assertThat(savedOrderTable).isNotNull();
+        assertThat(savedOrderTable.getNumberOfGuests()).isEqualTo(orderTable.getNumberOfGuests());
+    }
+
+    @Test
+    @DisplayName("주문 테이블 목록을 조회 할 수 있다.")
+    void list() {
+        final List<OrderTable> orderTables = Arrays.asList(orderTable);
+
+        given(orderTableDao.findAll()).willReturn(orderTables);
+
+        final List<OrderTable> list = tableBo.list();
+
+        assertThat(list).isNotEmpty();
+        assertThat(list.size()).isEqualTo(orderTables.size());
+    }
+
+    @Test
+    @DisplayName("주문 테이블의 상태를 비움으로 업데이트 할 수 있다.")
+    void changeEmpty() {
+        final OrderTable newOrderTable = orderTable;
+        newOrderTable.setTableGroupId(null);
+
+        given(orderTableDao.findById(orderTable.getId())).willReturn(Optional.of(orderTable));
+        given(orderDao.existsByOrderTableIdAndOrderStatusIn(orderTable.getId(),
+                Arrays.asList(OrderStatus.COOKING.name(), OrderStatus.MEAL.name()))).willReturn(false);
+        given(orderTableDao.save(orderTable)).willReturn(newOrderTable);
+
+        final OrderTable result = tableBo.changeEmpty(orderTable.getId(), newOrderTable);
+
+        assertThat(result.isEmpty()).isEqualTo(newOrderTable.isEmpty());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"-1"})
+    @DisplayName("주문 테이블이 정상 값이 아닌 경우")
+    void changeEmpty_exist_tableGroupId(final Long orderTableId) {
+        orderTable.setId(orderTableId);
+
+        assertThatExceptionOfType(IllegalArgumentException.class)
+                .isThrownBy(() -> tableBo.changeEmpty(orderTableId, orderTable));
+    }
+
+    @Test
+    @DisplayName("반드시 식사가 끝난 경우에 상태를 비움으로 업데이트 할 수 있다.")
+    void changeEmpty_check_orderStatus() {
+        final OrderTable resolvedOrderTable = orderTable;
+        resolvedOrderTable.setTableGroupId(null);
+
+        given(orderTableDao.findById(orderTable.getId())).willReturn(Optional.of(orderTable));
+        given(orderDao.existsByOrderTableIdAndOrderStatusIn(resolvedOrderTable.getId(),
+                Arrays.asList(OrderStatus.COOKING.name(), OrderStatus.MEAL.name()))).willThrow(IllegalArgumentException.class);
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
                Arrays.asList(OrderStatus.COOKING.name(), OrderStatus.MEAL.name()))).willReturn(true);
```

Reply:
👍